app/utils/AccountUtils.scala

```diff
@@ -22,17 +22,21 @@ import play.api.Play.current
 import play.api.i18n.Messages
 import play.api.i18n.Messages.Implicits._
 import play.api.mvc.Request
-import uk.gov.hmrc.domain.{SaUtr, TaxIdentifier, Uar}
+import uk.gov.hmrc.domain.{Nino, SaUtr, TaxIdentifier, Uar}
 
 trait AccountUtils {
-  def getAccount(request: AuthenticatedRequest[_]): TaxIdentifier = request.agentRef.getOrElse(request.saUtr.get)
+
+
+  def getAccount(request: AuthenticatedRequest[_]): TaxIdentifier = request.agentRef.getOrElse(request.saUtr.getOrElse(request.nino.get))
```

Comment:
This again seems to raise questions about SA vs PAYE precedence, and whether the behaviour is as expected for a user with both

Reply:
For each year it will be either PAYE or SA. And for a given year SA and SA Agent takes precedence over PAYE.